### DISCUSSION TITLE
Allow concurrent calls to prevent livewire from blocking itself

### DISCUSSION THREAD

**Discussion by eilerth** [livewire/livewire#6875] [1. Help]
**The Problem**
If you have multiple livewire components on a page, one will need to wait for the other before it can be processed.

In v2, this was done as separate requests back to the server, but in v3 it seems livewire will only have a single request back to the server at a time, and other interactions must wait for that request to complete before processing the next one. 

From what I've read, it sounds like this is by design due to a new feature where requests are bundled together, which makes sense in many scenarios.
However, in many other scenarios it causes a poor and sometimes confusing user experience.

**Suggestion**
It would be really great if there were a way to allow a component to make requests independent of other components. Perhaps a global default setting and/or a way to configure/override it on each individual livewire component.

**Setup for reproducing the problem**
Here are some steps to set up a scenario to see the behavior:

Create a few livewire components:
```
php artisan make:livewire MyComponent --inline
php artisan make:livewire MySlowToggle --inline
php artisan make:livewire MyFastToggle --inline
```

Add a route...
routes/web.php:
```
Route::get('/my-component', \App\Livewire\MyComponent::class);
```

app/Livewire/MyComponent.php:
```
class MyComponent extends Component
{
    public function render()
    {
		return <<<'HTML'
			<div>
				<div>1. Click the slow toggle checkbox and see it take 3 seconds to show "Checked" or "Not Checked" appropriately.</div>
				<div>2. Click the fast toggle checkbox several times and see it immediately show "Checked" or "Not Checked" appropriately.</div>
				<div>3. Click the slow toggle checkbox, immediately click the fast toggle checkbox, and see how it won't show "Checked" or "Not Checked" appropriately until the slow toggle component finishes.</div>
				<livewire:my-slow-toggle />
				<livewire:my-fast-toggle />
			</div>
		HTML;
    }
}
```

app/Livewire/MySlowToggle.php:
```
class MySlowToggle extends Component
{
	public $slowToggle = false;

    public function render()
    {
		sleep(3);
		return <<<'HTML'
		<div>
			Slow Toggle?
			<input type="checkbox" wire:model.live="slowToggle">
			{{ $slowToggle ? 'Checked' : 'Not Checked' }}
		</div>
		HTML;
    }
}
```

app/Livewire/MyFastToggle.php:
```
class MyFastToggle extends Component
{
	public $fastToggle = false;

    public function render()
    {
		return <<<'HTML'
		<div>
			Fast Toggle?
			<input type="checkbox" wire:model.live="fastToggle">
			{{ $fastToggle ? 'Checked' : 'Not Checked' }}
		</div>
		HTML;
    }
}
```

**Comment by joshhanley** [comment]
@eilerth we're already working on a solution for that, see PR #6819 🙂

**Reply by eilerth** [reply]
Well that produces a wonderful combination of relief, joy, and excitement. Thank you!

**Reply by nxgcx** [reply]
Hoping to see this functionality as well. Our project needs this.

**Reply by ErsanJSano** [reply]
Wondering on any timeline on this?! Thanks much!

**Reply by eilerth** [reply]
@joshhanley I took a deeper look at that PR and found it does not actually solve the problem I've described. However, it does address a very similar issue.

Ultimately I dove into the code and decided to submit another PR to solve the issue: https://github.com/livewire/livewire/pull/7102
